test/MetaTesting.lean

```diff
@@ -0,0 +1,381 @@
+import Mathlib.Algebra.Group.Nat
+import Mathlib.Tactic.MoveAdd
+import Mathlib.Tactic.Linter.MetaTesting
+
+section exclude
+
+/--
+info: Skipped since it contains 'guard_target'
+
+Use '#meta_test cmd' if you really want to run the test on 'cmd'
+-/
+#guard_msgs in
+set_option linter.metaTest true in
+/--
+info: Skipped since it contains 'guard_target'
+
+Use '#meta_test cmd' if you really want to run the test on 'cmd'
+-/
+#guard_msgs in
+example : Nat := by
+  guard_target = _
+  exact 0
+
+/--
+info: Skipped since it contains 'guard_target'
+
+Use '#meta_test cmd' if you really want to run the test on 'cmd'
+-/
+#guard_msgs in
+set_option linter.metaTest true in
+/--
+warning: missing withContext?
+
+  let _h : ?a✝ := _h
+  let _d : ?a✝¹ := _d
+  guard_target = _
+  exact 0
+  done
+---
+warning: missing withContext?
+
+  let _h : ?a✝ := _h
+  let _d : ?a✝¹ := _d
+  unfold_let  at *
+  guard_target = _
+  exact 0
+  done
```

Comment:
Why is this message appearing (almost) duplicated? and what does it tell us, that `exact` has no `withContext`?